src/js/components/DataTable/index.d.ts

```diff
@@ -1,53 +1,92 @@
-  primaryKey?: string | boolean;
-  replace?: boolean;
+    align?: 'center' | 'start' | 'end';
+    aggregate?: 'avg' | 'max' | 'min' | 'sum';
+    footer?: React.ReactNode | { aggregate?: boolean };
+    header?: string | React.ReactNode | { aggregate?: boolean };
+    primary?: boolean;
+    property: string;
+    render?: (datum: TRowType) => any;
+    search?: boolean;
+    sortable?: boolean;
+    size?: ColumnSizeType | string;
+    verticalAlign?: 'middle' | 'top' | 'bottom';
+  }[];
+  pad?: PadType | Sections<PadType>;
   resizeable?: boolean;
-  rowProps?: { [_:string]: { background?: {}, border?: {}, pad?: {}}};
-  size?: "small" | "medium" | "large" | "xlarge" | string;
-  sort?: { property: string, direction: "asc" | "desc" };
+  size?: 'small' | 'medium' | 'large' | 'xlarge' | string;
```

Comment:
I like the new structure you are offering, and I think the order within the structure should be alphabetical for consistency.
this line could be moved to line 66.